source/Octopus.Server.Client/OctopusServerEndpoint.cs

```diff
@@ -8,9 +8,31 @@ namespace Octopus.Client
     /// </summary>
     public class OctopusServerEndpoint
     {
-        readonly ILinkResolver octopusServer;
-        readonly string apiKey;
-        readonly ICredentials credentials;
+        /// <summary>
+        /// Create an instance with a Token to authenticate.
+        /// </summary>
+        public static OctopusServerEndpoint CreateWithToken(string octopusServerAddress, string token, ICredentials credentials = null)
```

Comment:
I have no issue with this but just wondering if this our standard approach (to have factory method)?

Reply:
Sorry I just posted this comment after I asked you to look at it so you probably didn't see it: https://github.com/OctopusDeploy/OctopusClients/pull/770#discussion_r1351341841